.github/workflows/release.yaml

```diff
@@ -28,13 +28,69 @@ jobs:
           check-latest: true
 
       - name: Run GoReleaser
+        id: run-goreleaser
         uses: goreleaser/goreleaser-action@f82d6c1c344bcacabba2c841718984797f664a6b # v4.2.0
         with:
           distribution: goreleaser
           version: latest
           args: release --clean
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          
+      - name: Generate subject
+        id: hash
+        env:
+          ARTIFACTS: "${{ steps.run-goreleaser.outputs.artifacts }}"
+        run: |
+          set -euo pipefail
+
+          checksum_file=$(echo "$ARTIFACTS" | jq -r '.[] | select (.type=="Checksum") | .path')
+          echo "hashes=$(cat $checksum_file | base64 -w0)" >> "$GITHUB_OUTPUT"
+  
+  provenance-for-binaries:
+    needs: [goreleaser]
+    permissions:
+      actions: read # To read the workflow path.
+      id-token: write # To sign the provenance.
+      contents: write # To add assets to a release.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.6.0
```

Comment:
Could we use the full hash changeset reference instead of the tag value here?

Reply:
hello @alacuku, unfortunately, we can't, it fails with the SHA256 reference because it builds some go binaries under the hood and they can work with the tag information only.

Reply:
... that was a great question! I had the same :D Thanks for the explanation :)